### PR TITLE
feat: add app insights sdk and feature toggle

### PR DESCRIPTION
**Pull request by JamesP-KC**
## Describe your changes

ASB-1971

- Add feature flag for enabling use of Application Insights
- Initialise Application Insights library on app startup if feature flag is enabled

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/applications-service-api/src/main.js [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
 
+const appInsights = require('applicationinsights');
+const config = require('./lib/config');
 const logger = require('./lib/logger');
 const server = require('./server');
 
 async function main() {
+	if (config.featureFlag.useApplicationInsights) {
+		try {
+			appInsights.setup().start();
```
Comment (Yakitrak) [inline packages/applications-service-api/src/main.js:11]: I am not familiar with azure insights but taking a look at the docs - do we not need to pass a connection string when doing `.setup()`?

Comment (Philip-Philippou1) [reply packages/applications-service-api/src/main.js:11]: Looks like appeals have activated with the following:

		appInsights.setup().setAutoDependencyCorrelation(true, true).setSendLiveMetrics(true);
		appInsights.defaultClient.context.tags[appInsights.defaultClient.context.keys.cloudRole] =
			'appeals-service-api';
		appInsights.start();
		
	I think the setAutoDependencyCorrelation is true by default, 
	the setSendLiveMetrics we may need to send the metrics to Azure,
	the  defaultClient.context.tags may be around splitting the insight by web app and api which may be useful.

Comment (JamesP-KC) [reply packages/applications-service-api/src/main.js:11]: The library should pick up the env var automatically

https://github.com/microsoft/ApplicationInsights-node.js#getting-started
> Configure the local SDK by calling appInsights.setup('YOUR_CONNECTION_STRING');, using the connection string you grabbed in step 2. Or put it in the APPLICATIONINSIGHTS_CONNECTION_STRING environment variable and call appInsights.setup() without parameters.

**Review by Yakitrak — APPROVED** [review]

**Pull request merged**